src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -58,103 +58,91 @@
+ * <h3>Locale Extensions</h3>
+ * <p>If the locale used for formatting contains "nu"
+ * (<a href="https://unicode.org/reports/tr35/#UnicodeNumberSystemIdentifier">Numbering System</a>)
+ * and/or "rg" (<a href="https://unicode.org/reports/tr35/#RegionOverride">Region Override</a>)
  * <a href="../util/Locale.html#def_locale_extension">Unicode extensions</a>,
  * the decimal digits, and/or the country used for formatting are overridden.
  * If both "nu" and "rg" are specified, the decimal digits from the "nu"
- * extension supersedes the implicit one from the "rg" extension.
+ * extension supersedes the implicit one from the "rg" extension. Additionally,
+ * currency formats support the "cf" ({@link #getCurrencyInstance(Locale) Currency
+ * Format style}) extension. Although the LDML specification defines various
+ * keys and values, actual locale-sensitive service implementations in a Java
+ * Runtime Environment might not support any particular Unicode locale attributes
+ * or key/type pairs.
+ * <p>Below is an example of a "en-US" locale with Thai digits,
+ * <blockquote>{@code NumberFormat.getInstance(Locale.forLanguageTag("en-US-u-nu-thai"));}</blockquote>
  *
- * <p>You can also use a {@code NumberFormat} to parse numbers:
- * <blockquote>
+ * <h2>Using NumberFormat</h2>
+ * The following is an example of formatting and parsing in a localized fashion,
  * {@snippet lang=java :
- * myNumber = nf.parse(myString);
+ * NumberFormat currencyFormat = NumberFormat.getCurrencyInstance(Locale.US);
+ * currencyFormat.format(100000); // returns "$100,000.00"
+ * currencyFormat.parse("$100,000.00"); // returns 100000
  * }
- * </blockquote>
- * Use {@code getInstance} or {@code getNumberInstance} to get the
- * normal number format. Use {@code getIntegerInstance} to get an
- * integer number format. Use {@code getCurrencyInstance} to get the
- * currency number format. Use {@code getCompactNumberInstance} to get the
- * compact number format to format a number in shorter form. For example,
- * {@code 2000} can be formatted as {@code "2K"} in
- * {@link java.util.Locale#US US locale}. Use {@code getPercentInstance}
- * to get a format for displaying percentages. With this format, a fraction
- * like 0.53 is displayed as 53%.
  *
- * <p>
- * You can also control the display of numbers with such methods as
- * {@code setMinimumFractionDigits}.
- * If you want even more control over the format or parsing,
- * or want to give your users more control,
- * you can try casting the {@code NumberFormat} you get from the factory methods
- * to a {@code DecimalFormat} or {@code CompactNumberFormat} depending on
- * the factory method used. This will work for the vast majority of locales;
- * just remember to put it in a {@code try} block in case you encounter
- * an unusual one.
+ * <h2>Customizing NumberFormat</h2>
+ * {@code NumberFormat} provides API to customize formatting and parsing behavior,
```

Comment:
Could make this PR a dependent PR of the leniency PR, and add `setStrict()` here as well.

Reply:
The leniency PR was integrated, I directly added setStrict() to this PR